test/e2e-test/app_rollout_test.go

```diff
@@ -0,0 +1,358 @@
+					return fmt.Errorf("pod status error %s", pod.Status.Phase)
+				}
+				if pod.Spec.Containers[0].Command[1] != fmt.Sprintf("stress-cpu=%s", cpu) {
+					return fmt.Errorf("pod cmmond haven't updated")
+				}
+			}
+			return nil
+		}, time.Second*120, time.Microsecond).Should(BeNil())
+	}
+
+	updateAppWithCpuAndPlan := func(app *v1beta1.Application, cpu string, plan *v1alpha1.RolloutPlan) {
+		Eventually(func() error {
+			checkApp := new(v1beta1.Application)
+			if err := k8sClient.Get(ctx, ctypes.NamespacedName{Namespace: namespaceName, Name: app.Name}, checkApp); err != nil {
+				return err
+			}
+			updateProperty := fmt.Sprintf(`{"cmd":["./podinfo","stress-cpu=%s"],"image":"stefanprodan/podinfo:4.0.3","port":8080,"replicas":6}`, cpu)
+			checkApp.Spec.Components[0].Properties.Raw = []byte(updateProperty)
+			checkApp.Spec.RolloutPlan = plan
+			if err := k8sClient.Update(ctx, checkApp); err != nil {
+				return err
+			}
+			return nil
+		}, time.Second*30, time.Microsecond*300).Should(BeNil())
+	}
+
+	It("Test upgrade application", func() {
+		plan := &v1alpha1.RolloutPlan{
+			RolloutStrategy: v1alpha1.IncreaseFirstRolloutStrategyType,
+			RolloutBatches: []v1alpha1.RolloutBatch{
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+			},
+			TargetSize: pointer.Int32Ptr(6),
+		}
+		appName = "app-rollout-1"
+		app := generateNewApp(appName, namespaceName, "clonesetservice", plan)
+		Expect(k8sClient.Create(ctx, app)).Should(BeNil())
+		verifyRolloutSucceeded(utils.ConstructRevisionName(appName, 1), "1")
+		updateAppWithCpuAndPlan(app, "2", plan)
+		verifyRolloutSucceeded(utils.ConstructRevisionName(appName, 2), "2")
+		updateAppWithCpuAndPlan(app, "3", plan)
+		verifyRolloutSucceeded(utils.ConstructRevisionName(appName, 3), "3")
+	})
+
+	It("Test application only upgrade batchPartition", func() {
+		plan := &v1alpha1.RolloutPlan{
+			RolloutStrategy: v1alpha1.IncreaseFirstRolloutStrategyType,
+			RolloutBatches: []v1alpha1.RolloutBatch{
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+			},
+			TargetSize: pointer.Int32Ptr(6),
+		}
+		appName = "app-roll-out-2"
+		app := generateNewApp(appName, namespaceName, "clonesetservice", plan)
+		Expect(k8sClient.Create(ctx, app)).Should(BeNil())
+		verifyRolloutSucceeded(utils.ConstructRevisionName(appName, 1), "1")
+		app.Spec.RolloutPlan.BatchPartition = pointer.Int32Ptr(0)
+		plan = &v1alpha1.RolloutPlan{
+			RolloutStrategy: v1alpha1.IncreaseFirstRolloutStrategyType,
+			RolloutBatches: []v1alpha1.RolloutBatch{
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+			},
+			TargetSize:     pointer.Int32Ptr(6),
+			BatchPartition: pointer.Int32Ptr(0),
+		}
+		updateAppWithCpuAndPlan(app, "2", plan)
+
+		By("upgrade first batch partition, verify the middle state")
```

Comment:
We don't really need to test the appRollout internal functionality here.  Just make sure that the application gets upgraded with the appRollout, especially the rollback/abandon/delete cases

Reply:
Here just  want to test update application really rollout batch by batch